src/main/java/org/osc/controller/nsc/utils/RedirectionApiUtils.java

```diff
@@ -168,18 +168,45 @@ public NetworkElementEntity txNetworkElementEntityById(Long id) {
         return this.em.find(NetworkElementEntity.class, id);
     }
 
-    public void removeSingleInspectionHook(InspectionHookEntity inspectionHookEntity) {
+    public void removeSingleInspectionHook(String hookId) {
+        InspectionHookEntity inspectionHook = this.txControl.required(() -> {
+            return this.em.find(InspectionHookEntity.class, hookId);
+        });
+
+        NetworkElementEntity inspected = inspectionHook.getInspectedPort();
```

Comment:
you should handle null here. If inspection hook is null, nothing to remove, just return